src/createServer.js

```diff
@@ -1,3 +1,61 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { convertToCase } = require('./convertToCase');
+const PORT = process.env.PORT || 5701;
+
+function createServer() {
+  return http.createServer((req, res) => {
+    const normalizedUrl = new URL(req.url, `http://localhost:${PORT}`);
+    const text = normalizedUrl.pathname.slice(1);
+    const toCase = normalizedUrl.searchParams.get('toCase');
+
+    const errors = [];
+
+    if (!text) {
+      errors.push({
+        message:
+          'Text to convert is required. Correct request is:' +
+          ' "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    }
+
+    if (!toCase) {
+      errors.push({
+        message:
+          '"toCase" query param is required. Correct request is: ' +
+          '"/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    }
+
+    const validCases = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+
+    if (toCase && !validCases.includes(toCase.toUpperCase())) {
+      errors.push({
+        message:
+          'This case is not supported. Available cases: ' +
+          'SNAKE, KEBAB, CAMEL, PASCAL, UPPER.',
```

Comment:
for better maintainability instead of using the hardcoded string "SNAKE, KEBAB, CAMEL, PASCAL, UPPER." you should use validCases array. This ensures that the list of valid cases is consistent and easier to update